src/api-wrappers/AXUIElement.swift

```diff
@@ -81,99 +85,16 @@ extension AXUIElement {
-        // Books.app has animations on window creation. This means windows are originally created with subrole == AXUnknown or isOnNormalLevel == false
-        return runningApp.bundleIdentifier == "com.apple.iBooksX"
-    }
-
-    private static func worldOfWarcraft(_ runningApp: NSRunningApplication, _ role: String?) -> Bool {
-        return runningApp.bundleIdentifier == "com.blizzard.worldofwarcraft" && role == kAXWindowRole
-    }
-
-    private static func battleNetBootstrapper(_ runningApp: NSRunningApplication, _ role: String?) -> Bool {
-        // Battlenet bootstrapper windows have subrole == AXUnknown
-        return runningApp.bundleIdentifier == "net.battle.bootstrapper" && role == kAXWindowRole
-    }
-
-    private static func drBetotte(_ runningApp: NSRunningApplication) -> Bool {
-        return runningApp.bundleIdentifier == "com.ssworks.drbetotte"
-    }
-
-    private static func dvdFab(_ runningApp: NSRunningApplication) -> Bool {
-        return runningApp.bundleIdentifier == "com.goland.dvdfab.macos"
-    }
-
-    private static func sanGuoShaAirWD(_ runningApp: NSRunningApplication) -> Bool {
-        return runningApp.bundleIdentifier == "SanGuoShaAirWD"
-    }
-
-    private static func steam(_ runningApp: NSRunningApplication, _ title: String?, _ role: String?) -> Bool {
-        // All Steam windows have subrole == AXUnknown
-        // some dropdown menus are not desirable; they have title == "", or sometimes role == nil when switching between menus quickly
-        return runningApp.bundleIdentifier == "com.valvesoftware.steam" && title != "" && role != nil
-    }
-
-    private static func firefoxFullscreenVideo(_ runningApp: NSRunningApplication, _ role: String?) -> Bool {
-        // Firefox fullscreen video have subrole == AXUnknown if fullscreen'ed when the base window is not fullscreen
-        return (runningApp.bundleIdentifier?.hasPrefix("org.mozilla.firefox") ?? false) && role == kAXWindowRole
-    }
-
-    private static func vlcFullscreenVideo(_ runningApp: NSRunningApplication, _ role: String?) -> Bool {
-        // VLC fullscreen video have subrole == AXUnknown if fullscreen'ed
-        return (runningApp.bundleIdentifier?.hasPrefix("org.videolan.vlc") ?? false) && role == kAXWindowRole
-    }
-
-    private static func androidEmulator(_ runningApp: NSRunningApplication, _ title: String?) -> Bool {
-        // android emulator small vertical menu is a "window" with empty title; we exclude it
-        return title != "" && Applications.isAndroidEmulator(runningApp)
+        return wid != 0 &&
+                size != nil && size!.width > minWindowSize && size!.height > minWindowSize &&
+                level == CGWindow.normalLevel &&
+                ([kAXStandardWindowSubrole, kAXDialogSubrole].contains(subrole) ||
+                        windowChecks.contains(where: { element in
+                            element.check(opts)
+                        }))
```

Comment:
This should have a similar short-circuiting effect compared to the old ORing method. If the checking order matters, it can be changed by rearranging the items in the `windowChecks` array.

Reply:
I think you lost some of the nuances of the checks in the process of uniform-ing the checks. Apps like Books and Keynote are at a [different level of the boolean chain than the apps later on](https://github.com/lwouis/alt-tab-macos/pull/1871/files#diff-ba0d5f463a7a746a839b98f1f7ab668395bfea51a4a29dad57dae4b26237b6c9L86). Same thing for Jetbrains.

Reply:
Oops, you're right. I added in an `isValidLevel()` function to `WindowCheck` in the latest push.